src/zenml/zen_server/models/component_models.py

```diff
@@ -0,0 +1,86 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Stack Models for the API endpoint definitions."""
+from typing import Any, Dict, Optional
+from uuid import UUID
+
+from pydantic import BaseModel, Field
+
+from zenml.enums import StackComponentType
+from zenml.models import ComponentModel
+
+
+class CreateComponentModel(BaseModel):
```

Comment:
Have  alook at how the other models have been updated - subclassing from ProjectScopedCreateRequest - we can hopwever also change this after the merge